app/src/main/java/fr/free/nrw/commons/upload/SingleUploadFragment.java

```diff
@@ -104,6 +105,16 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
 
         license = prefs.getString(Prefs.DEFAULT_LICENSE, Prefs.Licenses.CC_BY_SA_3);
 
+        // If this is a direct upload from Nearby, autofill title and desc fields with the Place's values
+        boolean isNearbyUpload = ((ShareActivity) getActivity()).isNearbyUpload();
+
+        if (isNearbyUpload) {
+            String imageTitle = ((ShareActivity) getActivity()).getNearbyTitle();
```

Comment:
Personally, I would move these to a (Dagger managed) shared data model class.  Even if you leave them based on shared preferences, I would move the members down here to get rid of the casting.  This is a symptom of [feature envy](http://wiki.c2.com/?FeatureEnvySmell).